pysolarmanv5/pysolarmanv5.py

```diff
@@ -302,13 +305,18 @@ def _data_receiver(self):
                 except ConnectionResetError:
                     self.log.debug(f"[{self.serial}] Connection RESET by peer.")
                     data = b""
+                except TimeoutError:
+                    self.log.debug(
+                        f"[{self.serial}] Sock.recv timed out ({self.socket_timeout} s)."
+                    )
+                    data = b""
```

Comment:
A generic exception handing can be added here in order the _reconnect() method to be called  always on error. Otherwise FD leaks still seem possible.

Reply:
Agree. I'll change the exception handling to a generic one since the handling is the same. The debug log message can differentiate errors from each other.